src/Spectre.AngularClient/src/app/divik-results/shared/divik.service.ts

```diff
@@ -24,53 +24,82 @@ import 'rxjs/Rx';
+  return <DivikConfig>({
+    maxK: json.MaxK,
+    level: json.Level,
+    usingLevels: json.UsingLevels,
+    amplitude: json.Amplitude,
+    variance: json.Variance,
+    percentSizeLimit: json.PercentSizeLimit,
+    featurePreservationLimit: json.FeaturePreservationLimit,
+    metric: json.Metric,
+    plottingPartitions: json.PlottingPartitions,
+    plottingPartitionsRecursively: json.PlottingPartitionsRecursively,
+    plottingDecomposition: json.PlottingDecomposition,
+    plottingDecompositionRecursively: json.PlottingDecompositionRecursively,
+    maxDecompositionComponents: json.MaxDecompositionComponents,
+  })
+    ;
 }
 
 function toHeatmap(response: Response): Heatmap {
-    console.log('[DivikService] parsing response:');
-    console.log(response);
-    const json = response.json();
-    console.log(json);
-    const max_column = Math.max.apply(null, json.X);
-    const min_column = Math.min.apply(null, json.X);
-    const max_row = Math.max.apply(null, json.Y);
-    const min_row = Math.min.apply(null, json.Y);
-    console.log('[DivikService] found bounds: (' + min_row + ', ' + max_row + '), (' + min_column + ', ' + max_column + ')');
-    console.log('[DivikService] initializing heatmap');
-    const data = [];
-    for (let i = min_row; i < max_row + 1; i++) {
-      data[i - min_row] = [];
-      for (let j = min_column; j < max_column + 1; j++) {
-        data[i - min_row][j - min_column] = 0;
-      }
-    }
-    console.log('[DivikService] processing heatmap');
-    for (let i = 0; i < json.X.length; i++) {
-        const column = json.X[i] - min_column;
-        const row = max_row - json.Y[i];
-        const divikData = json.Data[i];
-        console.log('[DivikService] processing (' + row + ', ' + column + ', ' + divikData + ')');
-        data[row][column] = divikData;
+  console.log('[DivikService] parsing response:');
+  console.log(response);
+  const json = response.json();
+  console.log(json);
+  const max_column = Math.max.apply(null, json.X);
+  const min_column = Math.min.apply(null, json.X);
+  const max_row = Math.max.apply(null, json.Y);
+  const min_row = Math.min.apply(null, json.Y);
+  console.log('[DivikService] found bounds: (' + min_row + ', ' + max_row + '), (' + min_column + ', ' + max_column + ')');
+  console.log('[DivikService] initializing heatmap');
+  const data = [];
+  for (let i = min_row; i < max_row + 1; i++) {
+    data[i - min_row] = [];
+    for (let j = min_column; j < max_column + 1; j++) {
+      data[i - min_row][j - min_column] = 0;
     }
-    console.log('[DivikService] heatmap processed');
-    return <Heatmap>({
-        data: data
-    });
+  }
+  console.log('[DivikService] processing heatmap');
+  for (let i = 0; i < json.X.length; i++) {
+    const column = json.X[i] - min_column;
+    const row = max_row - json.Y[i];
+    const divikData = json.Data[i];
+    console.log('[DivikService] processing (' + row + ', ' + column + ', ' + divikData + ')');
```

Comment:
in tslint.json we have no-console for console.info, console.debug so there is warning, change it in this file to allow debug?